pages/en/network/developing.mdx

```diff
@@ -0,0 +1,51 @@
+
+#### Deploy to the Subgraph Studio
+
+Once defined, the subgraph can be built and deployed to the [Subgraph Studio](https://thegraph.com/docs/en/deploying/subgraph-studio-faqs/). The Subgraph Studio is a sandbox environment which will index the deployed subgraph and make it available for rate-limited development and testing. This gives developers an opportunity to verify that their subgraph does not encounter any indexing errors, and works as expected.
+
+#### Publish to the Network
+
+When the developer is happy with their subgraph, they can publish it to The Graph Network. This is an on-chain action, which registers the subgraph so that it is discoverable by Indexers. Published subgraphs have a corresponding NFT, which is then easily transferable. The published subgraph has associated metadata, which provides other network participants with useful context and information.
+
+#### Signal to Encourage Indexing
+
+Published subgraphs are unlikely to be picked up by Indexers without the addition of signal. Signal is locked GRT associated with a given subgraph, which indicates to Indexers that a given subgraph will receive query volume, and also contributes to the indexing rewards available for processing it. Subgraph developers will generally add signal to their subgraph, in order to encourage indexing. Third party Curators may also signal on a given subgraph, if they deem the subgraph likely to drive query volume.
+
+#### Querying & Application Development
+
+Once a subgraph has been processed by Indexers and is available for querying, developers can start to use the subgraph in their applications. Developers query subgraphs via a gateway, which forwards their queries to an Indexer who has processed the subgraph, paying query fees in GRT.
+
+In order to make queries, developers must generate an API key, which can be done in the Subgraph Studio. This API key must be funded with GRT, in order to pay query fees. Developers can set a maximum query fee, in order to control their costs, and limit their API key to a given subgraph or origin domain. The Subgraph Studio provides developers with data on their API key usage over itme.
+
+Developers are also able to express an Indexer preference to the gateway, for example preferring Indexers whose query response is faster, or whose data is most up to date. These controls are set in the Subgraph Studio.
+
+#### Upgrading Subgraphs
+
+After a time a subgraph developer may want to update their subgraph, perhaps fixing a bug or adding new functionality. This requires another on-chain action to indicate the new version, which updates the specific subgraph manifest file associated with the subgraph. Upgrading the subgraph migrates any signal to the new version (assuming the user who applied the signal selected "auto-migrate"), which also incurs a migration tax. This signal migration should prompt Indexers to start indexing the new version of the subgraph, so it should soon become available for querying.
```

Comment:
The way this is written, it sounds like the subgraph manifest itself is updated. However - the subgraph manifest is deployed to IPFS and is immutable. 

When a new subgraph version is deployed, the manifest of the new version is deployed to IPFS, and an on-chain action indicates which deployment is the current version of the subgraph.

Maybe something like:
"After a time a subgraph developer may want to update their subgraph, perhaps fixing a bug or adding new functionality. The subgraph developer may deploy new version(s) of their subgraph to the `Subgraph Studio` for rate-limited development and testing. Once the Subgraph Developer is ready to upgrade, they can initiate a transaction to point their subgraph at the new version. Upgrading the subgraph ...

Reply:
Good call! Thank you - got that added in.